### PR TITLE
JOINDIN-600 Dont email author their own comments

### PR DESCRIPTION
**Pull request by exussum12**
This ensures that speakers do not get an email telling them they have
just replied.

The method is currently un tested, happy to add some but the function is
a mess too so would probably need refactoring. Another ticket ?

### CODE REVIEW

File: src/controllers/TalksController.php [hunk 1]
```diff
@@ -124,11 +124,17 @@ public function postAction($request, $db)
                         $comment    = $comment_mapper->getCommentById($new_id);
                         $speakers   = $talk_mapper->getSpeakerEmailsByTalkId($talk_id);
                         $recipients = array();
+                        $commentAuthorEmail = $comments[0]['email'];
```
Comment (heiglandreas) [inline src/controllers/TalksController.php]: Shouldn't that be ```$comment[0]['email']```?

File: src/controllers/TalksController.php [hunk 2]
```diff
@@ -124,11 +124,17 @@ public function postAction($request, $db)
...
                         foreach ($speakers as $person) {
+                            if ($commentAuthorEmail == $person['email']) {
+                                continue;
+                            }
                             $recipients[] = $person['email'];
                         }
-                        $emailService = new TalkCommentEmailService($this->config, $recipients, $talk, $comment);
-                        $emailService->sendEmail();
+                        if (count($recipients) > 0) {
+                            $emailService = new TalkCommentEmailService($this->config, $recipients, $talk, $comment);
```
Comment (heiglandreas) [inline src/controllers/TalksController.php]: So we could remove the ```if``` condition here.

File: src/controllers/TalksController.php (hunk 1 above)
Comment (exussum12) [inline src/controllers/TalksController.php]: Oops yes, This has been updated

File: src/controllers/TalksController.php (hunk 2 above)
Comment (exussum12) [inline src/controllers/TalksController.php]: removed from here and added to the send email

Comment (exussum12) [inline src/controllers/TalksController.php]: 2 mins, just checked further down stream, there is a foreach which will just not run if there are no emails

File: src/controllers/TalksController.php [hunk 3]
```diff
@@ -124,9 +124,14 @@ public function postAction($request, $db)
                         $comment    = $comment_mapper->getCommentById($new_id);
                         $speakers   = $talk_mapper->getSpeakerEmailsByTalkId($talk_id);
                         $recipients = array();
+                        $commentAuthorEmail = $comment[0]['email'];
```
Comment (heiglandreas) [inline src/controllers/TalksController.php]: Is this additional assignment necessary? As you're not calling a function there it should be possible to check against ```$comment[0]['email']``` on every foreach loop.

Comment (heiglandreas) [inline src/controllers/TalksController.php]: I'd say it's more readable without the additional assignment 😄 

Directly in the foreach I find a variable that I need to check what it contains before I can decide what it does. With the direct comparison I'd see it's both an email. 

Additionally the new variable is only used in this one single line. For me that usually is a sign that the assignment can be omitted.

Comment (exussum12) [inline src/controllers/TalksController.php]: without looking at the code around it theres no way to know what  `$comment[0]['email']` is. The variable makes that clear (and readable without having to lookup the meaning)
Happy to change if required

**Comment by liam-wiltshire** [comment]
Closed as duplicate.

**Pull request closed without merging**